core/domain/exp_domain.py

```diff
@@ -1259,6 +1259,36 @@ def update_interaction_confirmed_unclassified_answers(
         self.interaction.confirmed_unclassified_answers = (
             confirmed_unclassified_answers)
 
+    def can_undergo_classification(self):
+        """Checks if conditions for training are satisfied.
+
+        Args:
+            self: State. The state whose properties are to be checked for
+            training.
+
+        Returns:
+            bool: True, if the conditions are satisfied.
+        """
+        training_examples = [
+            [doc, []] for doc in (
+                self.interaction.confirmed_unclassified_answers)]
+        labels_ctr = 0
+        for (answer_group_index, answer_group) in enumerate(
+                self.interaction.answer_groups):
+            classifier_rule_spec_index = (
+                answer_group.get_classifier_rule_index())
+            if classifier_rule_spec_index is not None:
+                classifier_rule_spec = answer_group.rule_specs[
+                    classifier_rule_spec_index]
+                training_examples.extend([
+                    [doc, [str(answer_group_index)]]
```

Comment:
Hm, FYI, this implies that we'll need to reclassify / deactivate the model if e.g. a new answer group is added, since the indexes will change. Just noting this for future reference, unless we somehow want to bundle in additional instructions with the trained model to account for the reindexing.